Sprint 4/issue 21#add functionality for users to add new flight reservations and implement storage in firebase

Tole15 opened the pull request:
Successful Implementation of Flight Reservation Data, Autocomplete Picker and Data Storage

This update signifies the completion and successful operation of the flight reservation country data autocomplete picker feature along with the seamless storage of reservation information. The autocomplete picker has been integrated into the system, offering users an enhanced booking experience by simplifying the input of the country details and flight preferences. It intelligently suggests options as users type, making the process quicker and more convenient.

Furthermore, the implementation of data storage functionality ensures that users' flight reservation details are securely saved and readily accessible. By securely storing this information, users can easily manage their bookings, retrieve important details, and have a hassle-free travel planning experience.

Both the autocomplete picker and the reservation data storage are now fully operational, providing users with a comprehensive and efficient flight reservation system. These updates significantly enhance the overall user experience and contribute to a seamless and reliable flight booking process.

## Screenshots
![Screenshot_1689296365](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/129608820/a6a4fe8c-9875-4071-808f-c319570b16c7)
![Screenshot_1689296370](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/129608820/62542a3f-cec7-4b42-ac82-0d2ce470a720)
![Screenshot_1689296373](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/129608820/6cb0cbbd-83cf-4422-b568-df4a83fdccd3)
![Screenshot_1689296414](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/129608820/25dc8ab5-144c-4adc-8bd6-49ae168d6541)
![Screenshot_1689296417](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/129608820/9cbdfb8c-7bf8-4afd-a3eb-2601c457a01d)
![Screenshot_1689296359](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/129608820/cc0f66b0-41a9-4941-9d19-2570a3a24fc1)

Co-authored-by: BrandonVG <[brandon.vadillox3@gmail.com](mailto:brandon.vadillox3@gmail.com)>
Co-authored-by: Axw3ll <[axelomar.a.v@gmail.com](mailto:axelomar.a.v@gmail.com)>

Tole15 committed:
Display reservation info on card and add validation to 'Next' button.

Add functionality to display detailed reservation information of user's flight booking on the card while following the booking steps and implement validation for the 'Next' button to ensure all required fields are filled, effectively displaying an error message if any field is found to be empty.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
Refactor the PassengerPicker component to improve the handling of item selection and value handling.
- Added a new function, `handleItemSelected`, to handle the selection of an item from the wheel picker.
- Updated the `onItemSelected` prop of the WheelPicker component to use the `handleItemSelected` function.
- Created a new variable, `selectedIndex`, to store the index of the selected value in the `numbPassengers` array.
- Created a new variable, `selectedItem`, to determine the default selected item if the value is not found in the `numbPassengers` array.
- Updated the `selectedItem` prop of the WheelPicker component to use the `selectedItem` variable.

These changes ensure that the selected item is properly handled and that the component functions as expected.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
Implement separate style files in components lacking style separation
This commit introduces a crucial improvement by separating styles into separate files for components where this practice was not previously followed. By adopting this approach, we enhance code organization and maintainability while promoting a modular and reusable design pattern.
The changes made in this commit include:
Identifying components that did not have separate style files.
Creating individual style files for each of these components.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
Successful Implementation of Flight Reservation Data, Autocomplete Picker and Data Storage

This update signifies the completion and successful operation of the flight reservation country data autocomplete picker feature along with the seamless storage of reservation information. The autocomplete picker has been integrated into the system, offering users an enhanced booking experience by simplifying the input of the country details and flight preferences. It intelligently suggests options as users type, making the process quicker and more convenient.

Furthermore, the implementation of data storage functionality ensures that users' flight reservation details are securely saved and readily accessible. By securely storing this information, users can easily manage their bookings, retrieve important details, and have a hassle-free travel planning experience.

Both the autocomplete picker and the reservation data storage are now fully operational, providing users with a comprehensive and efficient flight reservation system. These updates significantly enhance the overall user experience and contribute to a seamless and reliable flight booking process.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

BrandonVG committed:
Update AppNavigation.jsx

Changed initial route 


Co-authored-by: Axw3ll <axelomar.a.v@gmail.com> 
Co-authored-by: Tole15 <rodrigermoon@gmail.com>

IvanCa27 reviewed (approved):
### _**Good job, guys**_

Eduardoespl reviewed (commented):
Good Job, lgtm!

Tole15 committed:
Code Format, Next Button Update & Validation Refactor.

We've executed a full code formatting using Prettier, improving readability across our project. Additionally, we have enabled the 'Next' button's disabled style until the current validation is complete, enhancing the user experience by preventing premature progress. In a move towards cleaner, more maintainable code, we've also separated the logic for input validation. This step was done with the intention to refactor the 'booking.jsx' screen and in keeping with best coding practices.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

Tole15 committed:
Added input validations, improved screen display, and updated README with project details.

-Implemented input validations in LogInScreen.
-Improved screen display and error handling.
-Imported Alert component in Booking.jsx to prevent errors.
-Updated README with project information and relevant images.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>

YilKanda commented on src/components/BackHeaderButton.jsx:
```diff
@@ -1,16 +1,22 @@
+      <Ionicons
+        name="airplane-sharp"
+        size={32}
+        color="#5974f5"
```
If you have a color library, you can use it here as well.

YilKanda commented on src/components/BookingInput.jsx:
```diff
@@ -1,12 +1,37 @@
+        dataSet={data}
+        textInputProps={{
+          placeholder: 'Select location',
+          style: {backgroundColor: '#FFFFFF'},
+        }}
+        rightButtonsContainerStyle={{backgroundColor: '#FFFFFF'}}
```
If you have a color library, you can use it here as well.

YilKanda commented on src/components/DatePickerCalendar.jsx:
```diff
@@ -1,94 +1,29 @@
+    <View style={DatePickerCalendarStyles.modalView}>
+      <DatePicker
+        options={{
+          mainColor: '#6172F6',
+          textFontSize: 17,
+          textHeaderFontSize: 20,
+          selectedTextColor: '#fff',
```
If you have a color library, you can use it here as well.

The styles must go in an external file.

YilKanda commented on src/data/Flights.json:
```diff
@@ -1,25 +1,124 @@
+      "pais": "United Arab Emirates"
+    }
+  ]
+}
```
Add a blank line

YilKanda commented on src/hooks/HandleLogin.js:
```diff
@@ -1,6 +1,20 @@
+    /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;
+
+  if (!emailRegex.test(email)) {
+    console.log('Invalid email format');
```
Do not leave debug code in production. You can use Alert.alert() instead.

YilKanda commented on src/hooks/HandleLogin.js:
```diff
@@ -1,6 +1,20 @@
+  }
+
+  if (!passwordRegex.test(password)) {
+    console.log('Incorrect email and/or password');
```
Do not leave debug code in production. You can use Alert.alert() instead.

YilKanda commented on src/hooks/SetFlights.js:
```diff
@@ -0,0 +1,19 @@
+      });
+    }
+  } catch (error) {
+    console.error('Error al agregar los valores a Firestore:', error);
```
Do not leave debug code in production. You can use Alert.alert() instead.

YilKanda commented on src/screens/Booking.jsx:
```diff
@@ -1,32 +1,175 @@
+          Nextclick();
+          setIsButtonDisabled(false);
+        } catch (error) {
+          console.error(error);
```
Do not leave debug code in production.

YilKanda commented on src/styles/AddButtonStyles.js:
```diff
@@ -0,0 +1,18 @@
+
+const AddButtonStyles = StyleSheet.create({
+  button: {
+    backgroundColor: '#5974f5',
```
If you have a color library use it.

YilKanda commented on src/styles/BookingInputStyles.js:
```diff
@@ -1,20 +1,12 @@
+  container: {
+    marginTop: 90,
     marginLeft: 21,
     borderBottomColor: '#bcbcbc',
```
If you have a color library use it.

YilKanda commented on src/styles/BookingStyles.js:
```diff
@@ -1,16 +1,24 @@
-    flex:1,
-    backgroundColor:'white',
+    flex: 1,
+    backgroundColor: 'white',
```
If you have a color library use it.

YilKanda commented on src/styles/NextButtonStyles.js:
```diff
@@ -1,29 +1,29 @@
-  export default NextButtonStyles;
\ No newline at end of file
+  button: {
+    backgroundColor: '#5974f5',
+    paddingVertical: 11,
+    paddingHorizontal: 10,
+    width: '90%',
+    borderRadius: 8,
+    elevation: 7,
+    shadowColor: '#0000ff',
+    shadowOffset: {width: 0, height: 2},
+    shadowOpacity: 0.3,
+    shadowRadius: 2,
```
Sort the styles alphabetically.

YilKanda commented on src/styles/NextButtonStyles.js:
```diff
@@ -1,29 +1,29 @@
+    shadowOpacity: 0.3,
+    shadowRadius: 2,
+  },
+  enabledBtn: {
+    backgroundColor: '#5974f5',
+  },
+  disabledBtn: {
+    backgroundColor: '#9e9e9e',
+  },
+  buttonText: {
+    color: '#fbfbfc',
```
If you have a color library, use it.

YilKanda commented on src/styles/NextButtonStyles.js:
```diff
@@ -1,29 +1,29 @@
+  },
+  buttonText: {
+    color: '#fbfbfc',
+    textAlign: 'center',
+    fontSize: 19,
+    fontWeight: 'bold',
```
Sort your styles alphabetically

BrandonVG committed:
Requested changes

made modifications to to comply with the requested changes

Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>
Co-authored-by: Tole15 <rodrigermoon@gmail.com>
